src/NodeTypeResolver/NodeTypeResolver.php

```diff
@@ -651,4 +659,20 @@ private function isEnumTypeMatch(MethodCall|NullsafeMethodCall $call, ObjectType
 
         return $classReflection->getName() === $objectType->getClassName();
     }
+
+    /**
+     * substr can return false on php 7.x and bellow
+     */
+    private function isSubstrOnPHP74(FuncCall $expr): bool
+    {
+        if ($expr->isFirstClassCallable()) {
+            return false;
+        }
+
+        if (! $this->nodeNameResolver->isName($expr, 'substr')) {
+            return false;
+        }
+
+        return ! $this->phpVersionProvider->isAtLeastPhpVersion(PhpVersion::PHP_80);
```

Comment:
PHPStan already detect it clearly when run on php 7.x

see https://phpstan.org/r/853f59a6-22ad-48ac-af93-66606b569613

on PHP 7.2 - 7.4 tab.

Reply:
But not on PHP 8.0+. This also allow to add our test with PHP 7.4 version.